packages/kbn-journeys/services/synthtrace.ts

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import {
+  ApmSynthtraceEsClient,
+  ApmSynthtraceKibanaClient,
+  InfraSynthtraceEsClient,
+  InfraSynthtraceKibanaClient,
+} from '@kbn/apm-synthtrace';
+import { ToolingLog } from '@kbn/tooling-log';
+import Url from 'url';
+import { Logger } from '@kbn/apm-synthtrace/src/lib/utils/create_logger';
+import { Auth, Es } from '.';
+import { KibanaUrl } from './kibana_url';
+
+export interface SynthtraceClientOptions {
+  kbnUrl: KibanaUrl;
+  auth: Auth;
+  es: Es;
+  log: ToolingLog;
+}
+export type SynthtraceClient = InfraSynthtraceEsClient | ApmSynthtraceEsClient;
+export type SynthtraceClientType = 'infra' | 'apm';
+
+export async function getSynthtraceClient(
+  type: SynthtraceClientType,
+  options: SynthtraceClientOptions
+): Promise<SynthtraceClient> {
+  if (type === 'infra') {
+    return initInfraSynthtraceClient(options);
+  } else {
+    return initApmSynthtraceClient(options);
+  }
+}
+
+// Adapting ToolingLog instance to Logger interface
+class LoggerAdapter implements Logger {
+  private log: ToolingLog;
+  private joiner = ', ';
+
+  constructor(log: ToolingLog) {
+    this.log = log;
+  }
+
+  debug(...args: any[]): void {
+    this.log.debug(args.join(this.joiner));
+  }
+
+  info(...args: any[]): void {
+    this.log.info(args.join(this.joiner));
+  }
+
+  error(arg: string | Error): void {
+    this.log.error(arg);
+  }
+
+  perf<T>(name: string, cb: () => T): T {
+    const startTime = Date.now();
+    const result = cb();
+    const duration = Date.now() - startTime;
+    const durationInSeconds = duration / 1000;
+    const formattedTime = durationInSeconds.toFixed(3) + 's';
+    this.log.info(`${name} took ${formattedTime}.`);
+    return result;
```

Comment:
This function is the reason to create Adapter. Not sure if it will be ever triggered, implementation is just something I quickly created myself.